src/icalendar/cal.py

```diff
@@ -436,6 +436,22 @@ def __repr__(self):
         subs = ', '.join(str(it) for it in self.subcomponents)
         return f"{self.name or type(self).__name__}({dict(self)}{', ' + subs if subs else ''})"
 
+    def __eq__(self, other):
```

Comment:
As a rule, overriding eq should also override hash. But I think, as long as the equality gets reduced, equality still implies hash equality.

Reply:
I think that implementing a `__hash__` for sub-class of CaselessDict is not straightforward as per [this](https://github.com/collective/icalendar/pull/391/files#r957502946}